bin/cli.ts

```diff
@@ -35,17 +53,24 @@ program
           console.error(`error: ${err.message}`)
         }
       }
-
-      // Exit with error code so CI fails
-      process.exit(1)
     }
 
-    const warns = results.filter((r) => r.type === 'warning')
     if (warns.length > 0) {
+      fs.appendFileSync(
+        validationResultsFilePath,
+        `These warnings were found during validation, but did not cause validation to fail:\n${warns
+          .map((warn) => warn.message)
+          .join('\r\n')}\n`
+      )
       for (const warn of warns) {
         console.log(`warning: ${warn.message}`)
       }
     }
+
+    if (errs.length > 0) {
+      // Exit with error code so CI fails
+      process.exit(1)
```

Comment:
nit unrelated to pr:

Would prefer if we just throw errors and then on the most outside external part of the code have some logic like

runValidaiton().catch(e => console.error('there was a problem', e) exit(1))